log4j-api:2.25.0 cannot (directly) be used with Java 8

## Description

log4j 2.25.0's dependency `com.github.spotbugs:spotbugs-annotations:4.9.3` is only compatible with JVM runtime version 11+

## Configuration

**Version:** 2.25.0

**Operating system:** Linux 6.14.6-2-MANJARO

**JDK:** Oracle Java 1.8.0

## Logs

from a `./gradlew build` with a `org.apache.logging.log4j:log4j-api:2.25.0` dependency using Java 8:

```
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve com.github.spotbugs:spotbugs-annotations:4.9.3.
     Required by:
         root project : > org.apache.logging.log4j:log4j-api:2.25.0
      > Dependency resolution is looking for a library compatible with JVM runtime version 8, but 'com.github.spotbugs:spotbugs-annotations:4.9.3' is only compatible with JVM runtime version 11 or newer.
```

## Reproduction

https://github.com/hankem/log4j-2.25.0-with-Java-8

--- a/.github/dependabot.yaml
+++ b/.github/dependabot.yaml
@@ -90,6 +90,9 @@ updates:
     registries:
       - maven-central
     ignore:
+      # `com.github.spotbugs:spotbugs-annotations:4.9.0` and onwards require Java 11
+      - dependency-name: "com.github.spotbugs:spotbugs-annotations"
+        versions: [ "[4.9.0,)" ]
       # Jetty 10.x does not have an internal logging API
       - dependency-name: "org.eclipse.jetty:*"
         versions: [ "[10,)" ]

--- a/log4j-parent/pom.xml
+++ b/log4j-parent/pom.xml
@@ -130,7 +130,8 @@
     <osgi.annotation.versioning.version>1.1.2</osgi.annotation.versioning.version>
     <pax-exam.version>4.14.0</pax-exam.version>
     <plexus-utils.version>3.6.0</plexus-utils.version>
-    <spotbugs-annotations.version>4.9.3</spotbugs-annotations.version>
+    <!-- `com.github.spotbugs:spotbugs-annotations:4.9.0` and onwards require Java 11: -->
+    <spotbugs-annotations.version>4.8.6</spotbugs-annotations.version>
     <spring-boot.version>2.7.18</spring-boot.version>
     <spring-framework.version>5.3.39</spring-framework.version>
     <system-stubs.version>2.0.3</system-stubs.version>

